### PR TITLE
refactor!: discontinue support for Minestom

### LINKED ISSUE #268: Remove Minestom platform
### Confirmation

- [X] I have checked for similar issues.

### Problem

Due to its artifact management, Minestom has been a pain to upkeep. In addition to this, it is very unlikely that someone is to use Chameleon for a Minestom extension - especially with the recent move to minestom-ce, which doesn't have extensions.

### Suggested solution

We should remove Minestom support altogether as it is unlikely to be used and no longer matches the purpose of Chameleon.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by joshuasing**
**Summary**
Discontinue support for Minestom and remove the Minestom Chameleon platform implementation.

Resolves #268

**Changes**
 - Discontinue support for Minestom
 - Remove Minestom Chameleon platform implementation (`dev.hypera:chameleon-platform-minestom`)
 - Remove unnecessary field from Sponge implementation (unrelated, spotted by errorprone)

**Checklist**
- [x] I acknowledge and agree to the terms of the [Developer Certificate of Origin](https://developercertificate.org/).
- [x] All contributed code can be distributed under the terms of the [MIT License](https://github.com/ChameleonFramework/Chameleon/blob/main/LICENSE).
- [x] I have read the [contributing guidelines](https://github.com/ChameleonFramework/Chameleon/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/ChameleonFramework/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have checked the ["Allow edit from maintainers"](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) option.
- [ ] I have added appropriate unit tests for my changes. 

**This pull request contains breaking changes.**  
This pull request should be merged with `[skip ci]` to prevent a release without a version bump.

### CODE REVIEW

File: platform-sponge/src/main/java/dev/hypera/chameleon/platform/sponge/SpongeChameleon.java [hunk 1]
```diff
@@ -73,7 +72,6 @@ public final class SpongeChameleon extends PlatformChameleon<SpongePlugin> {
         @NotNull ExtensionMap extensions
     ) throws ChameleonInstantiationException {
         super(chameleonPlugin, spongePlugin, eventBus, logger, extensions);
-        this.plugin = spongePlugin;
```
Comment (LooFifteen) [inline platform-sponge/src/main/java/dev/hypera/chameleon/platform/sponge/SpongeChameleon.java:76]: What's the reason for this change?

**Review by LooFifteen — APPROVED** [review]

File: platform-sponge/src/main/java/dev/hypera/chameleon/platform/sponge/SpongeChameleon.java (hunk 1 above)
Comment (joshuasing) [reply platform-sponge/src/main/java/dev/hypera/chameleon/platform/sponge/SpongeChameleon.java:76]: `SpongeChameleon` extends `PlatformChameleon<SpongePlugin>`, which provides `plugin` as a protected field (and provides `#getPlatformPlugin()`)
I made this change to resolve a build warning from errorprone that mentioned this field was shadowing the protected field and could lead to confusion.

This change should have been made in https://github.com/ChameleonFramework/Chameleon/commit/18d400201198a8f57815c8e9b16945e4bb998fb8 but got left out.
